tests/testthat/test-summary-log.R

```diff
@@ -0,0 +1,97 @@
+context("Test creating summary logs")
+
+if (Sys.getenv("METWORX_VERSION") == "" && Sys.getenv("DRONE") != "true") {
+  skip("test-summary only runs on Metworx or Drone")
+}
+
+# references
+OFV_REF <- 2636.846
+PARAM_COUNT_REF <- 7
+
+# helper to run expectations
+test_sum_df <- function(sum_df, .paths, .col_count) {
+  num_mods <- length(.paths)
+  expect_equal(nrow(sum_df), num_mods)
+  expect_equal(ncol(sum_df), .col_count)
+
+  # check some columns
+  expect_equal(sum_df$absolute_model_path, purrr::map_chr(.paths, ~normalizePath(.x)))
+  expect_true(all(is.na(sum_df$error_msg)))
+  expect_false(any(sum_df$needed_fail_flags))
+  expect_equal(sum_df$ofv, rep(OFV_REF, num_mods))
+  expect_equal(sum_df$param_count, rep(PARAM_COUNT_REF, num_mods))
+  expect_false(any(sum_df$minimization_terminated))
+}
+
+setup({
+  cleanup()
+  create_all_models()
+  fs::dir_copy(MOD1_PATH, NEW_MOD2)
+  fs::dir_copy(MOD1_PATH, NEW_MOD3)
+  fs::dir_copy(MOD1_PATH, LEVEL2_MOD)
+})
+teardown({
+  cleanup()
+})
+
+withr::with_options(list(rbabylon.model_directory = NULL), {
+
+  test_that("summary_log() errors with malformed YAML", {
+    log_df <- expect_error(summary_log(), regexp = "Unexpected error.+model_path defined in yaml")
+  })
+
+  test_that("summary_log() returns NULL and warns when no YAML found", {
+    log_df <- expect_warning(summary_log("data"), regexp = "Found no valid model YAML files in data")
+    expect_true(is.null(log_df))
+  })
+}) # closing withr::with_options
+
+
+withr::with_options(list(rbabylon.bbi_exe_path = read_bbi_path(),
+                         rbabylon.model_directory = normalizePath(MODEL_DIR)), {
+
+  #########################################
+  # extracting things from summary object
+  #########################################
+
+  test_that("summary_log() works correctly with nested dirs", {
```

Comment:
As with `config_log()`, this is a consequence of `find_models()`.

Reply:
same https://github.com/metrumresearchgroup/rbabylon/pull/118/files#r466543858